lib/oli_web/live/workspaces/student.ex

```diff
@@ -429,8 +416,38 @@ defmodule OliWeb.Workspaces.Student do
     }
   end
 
-  defp user_is_only_a_student?(%{can_create_sections: true}, _sections), do: false
+  defp user_is_only_a_student?(%{can_create_sections: true}), do: false
+
+  defp user_is_only_a_student?(%{id: user_id}) do
+    !Repo.exists?(
+      from e in Enrollment,
+        left_join: ecr in EnrollmentContextRole,
+        on: e.id == ecr.enrollment_id,
+        left_join: upr in "users_platform_roles",
+        on: e.user_id == upr.user_id,
+        where: e.user_id == ^user_id,
+        where:
+          ecr.context_role_id not in ^@context_student_roles_ids or
+            upr.platform_role_id not in ^@platform_student_roles_ids
```

Comment:
I wonder if this part of the query works as we expect. Let's say a student has only a student platform role but **not** a student context role. In that case, line `430` will return `true` regardless that the user is platform student, and the whole query will return `false` which I understand is not the desired outcome. I think changing the `or` with an `and` would fix this edge case. Wdyt?

Reply:
Agree, we should simply not be considering platform roles in the query

```
 from e in Enrollment,
        left_join: ecr in EnrollmentContextRole,
        on: e.id == ecr.enrollment_id and e.user_id == ^user_id,
        where:
          ecr.context_role_id not in ^@context_student_roles_ids
```

Reply:
Thanks for the input!
Now the query is:
```elixir
      from sections in Section,
        join: enrollments in assoc(sections, :enrollments),
        join: context_roles in EnrollmentContextRole,
        on: enrollments.id == context_roles.enrollment_id,
        where: enrollments.user_id == ^user_id,
        where: sections.open_and_free == true,
        where: sections.status == :active,
        where: context_roles.context_role_id in ^context_role_ids,
        select: sections

```